CRM/Utils/System/Drupal8.php

```diff
@@ -554,7 +554,14 @@ public function getModules() {
     $module_data = \Drupal::service('extension.list.module')->reset()->getList();
     foreach ($module_data as $module_name => $extension) {
       if (!isset($extension->info['hidden']) && $extension->origin != 'core') {
-        $extension->schema_version = drupal_get_installed_schema_version($module_name);
+        // @TODO This isn't the same as the Drupal 7 version. In 8 the status
+        // field means installed or not - there is no disabled, so this is
+        // including ALL non-core modules and then for the isEnabled parameter
+        // it's actually indicating whether it's installed or not which isn't
+        // quite the same. This seems really only used by managed entities, so
+        // if someone wants to look into (a) why is managed entities looking at
+        // drupal modules, and (b) what does it actually need about them, then
+        // could update this.
```

Comment:
> (a) why is managed entities looking at drupal modules, and (b) what does it actually need about them

__TLDR__: Yeah, I don't think it really needs the schema version per se - it needs to distinguish modules which are installed/disabled/uninstalled, ie

__Short Enough/Will Read__ (SEWR?): [`hook_civicrm_managed`](https://docs.civicrm.org/dev/en/latest/hooks/hook_civicrm_managed/) can be implemented by CMS modules (although it's not very common - and it takes a little boilerplate).  Adapting the example from docs.c.o to a Drupal module (say, `webform_civicrm`), it would look like this:

```php
function webform_civicrm_civicrm_managed(&$entities) {
  $entities[] = array(
    'module' => 'drupal.webform_civicrm',
    'name' => 'myreport',
    'entity' => 'ReportTemplate',
    'params' => array(...),
  );
}
```

Note the origin `module`. `CRM_Core_ManagedEntities` uses the status of `drupal.webform_civicrm` to decide between whether to `INSERT`/`UPDATE`, whether to deactivate (`UPDATE ... SET is_active=0`), or whether to remove (`DELETE`) the `myreport` entity.

To my knowledge, the old line is superfluous. The next line (`new CRM_Core_Module(...($extension->status == 1))`) is what matters.

Reply:
Thanks that's helpful info - I wasn't expecting anyone to respond. I think I will make a separate PR though, because I don't want the part I care about (removing the deprecation) to get held up by having to review changes to managed entities in drupal modules.